### ISSUE TITLE
Reset stack tool window level after adjustment

### ISSUE THREAD

**Issue by kofifus** [cornerstonejs/cornerstoneTools#209]
The viewport's stack maintains the WL width/center values across all images in the stack.

If the window level was **not** manually changed (ie with the WL tool), will every image in the viewport's stack display with WL values from it's dataset ? or will all images use the WL values of the first image ?

If the latter then this is an issue and I need some pointers into how to solve it ..

Further, after manually changing WL  (ie with the WL tool), how can I revert back to the unmodified values ? (ie on a 'reset' button click), that is go back to each image in the stack using the WL from it's dataset ?

Thx!

**Comment by dannyrb** [comment]
@kofifus, I don't know the answer to your first question, but you should be able to test it with a stack that has mixed window/level values. My best guess is that the stack will use the WL values from the first image loaded.

I see that I manually call 

To reset levels to the default value for the current image, I use something similar to the following snippet:
```
      const canvas = document.getElementById("my-cornerstone-canvas")

      const image = cornerstone.getImage(canvas)
      const defaultViewport = cornerstone.getDefaultViewportForImage(canvas, image)

      let viewport = cornerstone.getViewport(canvas)
      viewport.voi.windowWidth = defaultViewport.voi.windowWidth
      viewport.voi.windowCenter = defaultViewport.voi.windowCenter
      cornerstone.setViewport(canvas, viewport)
```

When I load a new stack/series, I call:
`cornerstone.reset(activeCanvas)`

Let me know if you have any follow-up questions. If you feel we should add documentation or a "code recipe" to the repository, please let me know! If this solves your problem, please mark this as closed (:

**Comment by kofifus** [comment]
Please leave the issue open as the first question is important to me and still unanswered ...

Thx for the code ! I'll give it a try ...

**Comment by dannyrb** [comment]
This has been open for a while. I'm going to close due to inactivity. You should be able to answer your question by creating a simple POC that has images with different WW/WC values and loading one after the other.

**Comment by kevinleedrum** [comment]
We ran into the issue that @kofifus described.  Some image stacks will have varying default WW/WC values between images, however the cornerstone behavior is to reuse the WW/WC values between images in a stack.  Most other DICOM viewers only reuse the WW/WC values across a stack once a manual adjustment has been made.  I would consider creating an issue for this, but I would worry that this would be too much of a breaking change, and the "correct" behavior is probably arguable.  It might be better to just work around as needed for anyone who wants the behavior of other viewers.

For posterity, the workaround I came up with:
- In our image viewer, we have an `isAnyImageLevelAdjusted` flag that defaults to false, meaning no manual adjustments have been made.
- We also reset this flag to false any time the image is `cornerstone.reset` or whenever we switch to a different image stack.
- Inside an event listener attached to our canvas for the `cornerstone.EVENTS.NEW_IMAGE` event, if that flag is false, we reset the WW/WC values.  Something like `event.detail.viewport.voi = { ...cornerstone.getDefaultViewportForImage(canvas, event.detail.image).voi }`.
- Inside an event listener for `cornerstone.EVENTS.IMAGE_RENDERED`, we set the `isAnyImageLevelAdjusted` flag to true if the WW/WC values are not the defaults (e.g. `if (event.detail.viewport.voi.windowWidth !== defaultViewport.voi.windowWidth) this.isAnyImageLevelAdjusted = true`).  Now that the flag is set to true, we allow cornerstone to do its default behavior of reusing these adjusted WW/WC values.
